### ISSUE TITLE
Segfault when opening a stream

### ISSUE THREAD

**Issue by ptkato** [vinszent/gnome-twitch#331]
Any stream that I try to open crashes GT, the terminal output has nothing relevant. I'm using b5fa8a4.

dmesg output:

    [30437.968844] pool[626]: segfault at 0 ip           (null) sp 00007f837f092b68 error 14 in gnome-twitch[55b50859c000+7c000]
    [30437.968859] gnome-twitch[611]: segfault at 0 ip           (null) sp 00007fff738e3908 error 14
    [30437.968868]  in gnome-twitch[55b50859c000+7c000]

**Comment by ptkato** [comment]
I did some further testing, it only happens when using GStramer Clutter player backend.

**Comment by vinszent** [comment]
Yeah I haven't updated the Clutter backend to use the new API. TBH I would like to drop the Clutter one because I don't see what the Clutter one does that the normal OpenGL doesn't do already.

**Comment by ptkato** [comment]
Now doesn't matter the backend, it randomly crashes, usually in the third try it doesn't crash.

```
[  591.814632] pool[1544]: segfault at 0 ip           (null) sp 00007fe651777b68 error 14 in gnome-twitch[55db5b43b000+7c000]
```

**Comment by vinszent** [comment]
Can you please build with debugging symbols, so I can see where it crashes?

**Comment by ptkato** [comment]
Whaddaya mean? Can't I just run with --log-level=debug, or were the symbols removed from the git version?

**Comment by TingPing** [comment]
> Yeah I haven't updated the Clutter backend to use the new API. TBH I would like to drop the Clutter one because I don't see what the Clutter one does that the normal OpenGL doesn't do already.

Bit off topic but personally I think gnome-twitch should drop the multiple backend mess and just pick a single high quality and well tested one. Clutter is basically dead, the MPV backend is too large of a dependency IMO, which leaves Cairo and OpenGL. The latter is certainly the future of Gtk/GStreamer but in my experience the Cairo one is still more stable atm.

**Comment by vinszent** [comment]
Yep I will be dropping the Clutter backend as stated before. I will also be merging the Cairo and OpenGL backends into one, as their dependencies are the same. Then the user can go into the plugin settings and pick which type of rendering they want. 

However I still want to keep the MPV backend as a number of people use it and if you don't have a GNOME based desktop, GStreamer brings in tonnes of dependencies.
